File: app/src/main/java/ca/unb/mobiledev/ultimatestattracker/model/Team.java
package ca.unb.mobiledev.ultimatestattracker.model;

import java.io.Serializable;
import java.util.ArrayList;

public class Team implements Serializable {
    public String teamName;
    public ArrayList<Player> players;

    /* Constructors */
    public Team(String name, ArrayList<Player> players){
        this.teamName = name;
        this.players = players;
    }
    public Team(String name, Player player){
        this.teamName = name;
        ArrayList<Player> list = new ArrayList<Player>();
        list.add(player);
        this.players = list;
    }
    public Team(String name){
        this.teamName = name;
        ArrayList<Player> list = new ArrayList<Player>();
        this.players = list;
    }
}

File: app/src/main/java/ca/unb/mobiledev/ultimatestattracker/model/Team.kt
package ca.unb.mobiledev.ultimatestattracker.model

import android.content.Context
import android.content.Context.MODE_PRIVATE
import android.widget.Toast
import ca.unb.mobiledev.ultimatestattracker.helper.FileUtils.getTeamDir
import ca.unb.mobiledev.ultimatestattracker.helper.FileUtils.toFileName
import ca.unb.mobiledev.ultimatestattracker.helper.JsonUtils
import java.io.File

class Team : java.io.Serializable {
    var teamName: String
    var players: ArrayList<Player>

    constructor(name: String, players: ArrayList<Player>) {
        this.teamName = name
        this.players = players
    }

    fun addPlayer(player: Player, c: Context) : Boolean {
        for(p in players){
            if(p.number == player.number){
                Toast.makeText(c, "Player with that number already exists", Toast.LENGTH_SHORT).show()
                return false
            }
        }
        this.players.add(player)
        return true
    }

    fun removePlayer(player: Player) {
        this.players.remove(player)
    }

    fun save(c : Context) {
        val json = JsonUtils()
        val teamDir = getTeamDir(teamName, c)
        val fileName = "${toFileName(teamName)}.json"
        val file = File(teamDir, fileName)
        val jsonStr = json.convertObjectToJson(this)
        if (jsonStr != null)
            file.writeText(jsonStr)
    }

    override fun toString() : String {
        return "Team Name: ${this.teamName} - Players: ${this.players}"
    }
}